cmd/tendermint/commands/reset_priv_validator.go

```diff
@@ -28,25 +30,38 @@ func init() {
 // XXX: this is totally unsafe.
 // it's only suitable for testnets.
 func resetAll(cmd *cobra.Command, args []string) {
-	resetPrivValidator(cmd, args)
-	os.RemoveAll(config.GetString("db_dir"))
-	os.Remove(config.GetString("cs_wal_file"))
+	ResetAll(config, log)
 }
 
 // XXX: this is totally unsafe.
 // it's only suitable for testnets.
 func resetPrivValidator(cmd *cobra.Command, args []string) {
+	ResetPrivValidator(config, log)
+}
+
+func ResetAll(c *viper.Viper, l log15.Logger) {
+	ResetPrivValidator(c, l)
+	dataDir := c.GetString("db_dir")
+	walFile := c.GetString("cs_wal_file")
+	os.RemoveAll(dataDir)
+	os.Remove(walFile)
+	l.Notice("Removed all data", "dir", dataDir)
+	l.Notice("Removed walfile", "file", walFile)
+}
+
+func ResetPrivValidator(c *viper.Viper, l log15.Logger) {
+
 	// Get PrivValidator
 	var privValidator *types.PrivValidator
-	privValidatorFile := config.GetString("priv_validator_file")
+	privValidatorFile := c.GetString("priv_validator_file")
 	if _, err := os.Stat(privValidatorFile); err == nil {
 		privValidator = types.LoadPrivValidator(privValidatorFile)
 		privValidator.Reset()
-		log.Notice("Reset PrivValidator", "file", privValidatorFile)
+		l.Notice("Reset PrivValidator", "file", privValidatorFile)
 	} else {
 		privValidator = types.GenPrivValidator()
 		privValidator.SetFile(privValidatorFile)
 		privValidator.Save()
-		log.Notice("Generated PrivValidator", "file", privValidatorFile)
+		l.Notice("Generated PrivValidator", "file", privValidatorFile)
```

Comment:
The naming was changed to avoid confusion with the global variables `config` and `log` -  these are now local variables in these functions